python/ctsm/test/test_unit_gen_mksurfdata_jobscript_single.py

```diff
@@ -0,0 +1,251 @@
+#!/usr/bin/env python3
```

Comment:
@samsrabin this is the new unit tester I added that I'd like to go over.

Reply:
Some suggestions here were to move the expected value setting from Setup to where it's actually used.

Some names of tests could be better to show they are failure tests.
There is a utility function that could be moved out.

Reply:
In issues #2446 and #2439